force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,136 @@
+public class OpportunityTriggerHandler extends TriggerHandler {
+    
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id,Opportunity> newOppMap;
+    private Map<Id,Opportunity> oldOppMap;
+
+    public OpportunityTriggerHandler() {
+        
+        this.newOppList = (List<Opportunity>)Trigger.new;
+        this.oldOppList = (List<Opportunity>)Trigger.old;
+        this.newOppMap = (Map<Id,Opportunity>)Trigger.newMap;
+        this.oldOppMap = (Map<Id,Opportunity>)Trigger.oldMap;
+        //this.setMaxLoopCount(1);
+
+    }
+    
+    public override void beforeInsert() {
+
+        // Set default Type for new Opportunities            
+        for (Opportunity Opp : this.newOppList) {
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }
+        }        
+    }
+
+    public override void beforeUpdate() {
+
+        // *** Validate that the amount is greater than 5000.
+        
+        for(Opportunity opp : this.newOppList) {
+            if(opp.Amount <= 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+
+        // *** Set the primary contact on the opportunity to the contact with the title of 'CEO'.
+    
+        // Get contacts related to the opportunity account
+        Set<Id> accountIds = new Set<Id>();
+        for(Opportunity opp : this.newOppList) {
+            accountIds.add(opp.AccountId);
+        }
+        
+        Map<Id, Contact> contacts = new Map<Id, Contact>([
+            SELECT Id, FirstName, AccountId FROM Contact 
+            WHERE AccountId IN :accountIds AND Title = 'CEO' ORDER BY FirstName ASC]);
+        Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+
+        for (Contact cont : contacts.values()) {
+            if (!accountIdToContact.containsKey(cont.AccountId)) {
+                accountIdToContact.put(cont.AccountId, cont);
+            }
+        }
+
+        for(Opportunity opp : this.newOppList){
+            if(opp.Primary_Contact__c == null){
+                if (accountIdToContact.containsKey(opp.AccountId)){
+                    opp.Primary_Contact__c = accountIdToContact.get(opp.AccountId).Id;
+                }
+            }
+        }
+    }
```

Comment:
You can see where this will get harder to test and maintain. This method is doing two things now. Validating the min value and  setting a contact